libraries/standard/http/test/test-HTTPClient_InitializeRequestHeaders.c

```diff
@@ -0,0 +1,189 @@
+#include <string.h>
+
+#include "common.h"
+
+/* Functions are pulled out into their own C files to be tested as a unit. */
+#include "_writeRequestLine.c"
```

Comment:
Will all embedded compilers allow this?

Reply:
We already #include c source files in the current MQTT tests that run on all of our platforms.

Reply:
We are going to convert these tests to the cmock framework and unity after CMAKE is in. For now we are writing tests in what is available. It is an easy task to convert our tests.

Reply:
The OTA library implementation did this (i.e. include C modules) at some point and we ran into problems, both with subsequent ports and with subsequent code-review visibility. I think we should find a different approach.

Reply:
We are going to convert these in the future. These WILL be gone. For now we are using https://github.com/dan4thewin/mock4thewin for unit testing.

These includes are only for Dan's testing framework which builds each function separately with their dependent private function for mocking. We do not have Unity in the development branch yet and moving these tests is a separate activity which will be easy to do.

Reply:
I think we can resolve this for now as it will be corrected as part of the conversion. The issues we ran into in the past was because we included a parameterized file which the name was passed into the compiler on the command line, that is not supported by all, but what is used here is valid C according to the standard on all C compilers (as stated at  section 6.10.2 of the C standard, it is entitled "Source file inclusion" in the spec,  to be 100% clear the description in the standard states "A #include directive shall identify a header or source file that can be processed by the
implementation)" and so forth. 

Let's resolve this one for now.